exercises/security-rnd/software-engineer/README.md

```diff
@@ -20,20 +20,23 @@ You can obtain package data through the npm registry: `https://registry.npmjs.or
 
 #### Things to consider
 
-  1. Look at the inner "dependencies" object for analysis of first-order dependencies.
+  1. Look at the inner "dependencies" object for analysis of its dependencies and do so recursively.
```

Comment:
I'm re-reading this compared to the original and can see what the original author meant. But also how it can be confusing.

Reply:
After reading it 5 times, I can't understand the original meaning 😅

Reply:
Basically - use npms API response to get the first level of dependencies. The implication was that you should then use that to get the next level recursively. It was just never explicitly said.

Reply:
By explicitly saying recursive, i think we kind of missing the point of the test. Maybe we can go with something a bit more vague, like:
```suggestion
  1. Look at the inner "dependencies" object for analysis of all of its dependencies.
```

Reply:
I think this is the part of the exercise we should be a bit more explicit about. It's confused literally every single applicant to the groot pipeline we had.

It seems to be particularly easy to misunderstand if English is not your first language

Reply:
I definitely agree this is the most important part to improve and make a bit more explicit, but mentioning `recursive` makes this a no brainer to how to implement. I think we should try to emphasise the needs/expected outcome much clearer, but leave the implementation a bit vague to have some differences between candidates, specifically about how they interpreted it.